crates/ruff_python_parser/src/parser/helpers.rs

```diff
@@ -123,26 +123,3 @@ pub(super) fn validate_parameters(parameters: &ast::Parameters) -> Result<(), Pa
 
     Ok(())
 }
-
-pub(super) fn validate_arguments(arguments: &ast::Arguments) -> Result<(), ParseError> {
-    let mut all_arg_names = FxHashSet::with_capacity_and_hasher(
-        arguments.keywords.len(),
-        BuildHasherDefault::default(),
-    );
-
-    for (name, range) in arguments
-        .keywords
-        .iter()
-        .filter_map(|argument| argument.arg.as_ref().map(|arg| (arg, argument.range)))
-    {
-        let arg_name = name.as_str();
-        if !all_arg_names.insert(arg_name) {
-            return Err(ParseError {
-                error: ParseErrorType::DuplicateKeywordArgumentError(arg_name.to_string()),
-                location: range,
-            });
-        }
-    }
-
-    Ok(())
-}
```

Comment:
This is a soft syntax error so remove it from the parser.

Reply:
> What's a soft syntax error? Is it one that is only thrown at runtime (but still considered a `SyntaxError`)?

Yes, that's correct.

> Changing this could also break downstream rules that incorrectly assume that keyword arguments are guaranteed to be unique (for a valid AST)

Oh right, thanks for noticing this.

Reply:
Ok, it turns out that the validation function short circuits on the first error. I'll expand it to return a vector of `ParseError` instead.